### ISSUE TITLE
[iOS] error: failed to emit precompiled header Runner-Bridging-Header-swift_1IAYIT96INIUM-clang_3MJ7WMOU6EFXF.pch' for bridging header

### ISSUE THREAD

**Issue by andreibratuleanu21** [flutter/flutter#96800]
Saw many posts with this issue but none solved my problem. I'm trying to create a flutter package / library with Kotlin and Swift.

## Steps to Reproduce

Not much to say here. Happend after latest flutter upgrade.

Things that I tried:
- Flutter clean, remove symlinks, pod deintegrate, rm Podfile.lock and even`rm -rf ios` & run flutter create again.
- In Xcode, clean build & checked Targets > Runner > Build Phases (also Signing and other settings)
- Downgrade flutter to v2.5
- Tried a different device

**Expected results:** flutter run should build

**Actual results:** flutter run fails to compile with:
```
[package_name]/example/ios/Runner/GeneratedPluginRegistrant.h:10:9: error: 'Flutter/Flutter.h' file not found
...
<unknown>:0: error: failed to emit precompiled header
```

<details>
  <summary>Logs</summary>

`flutter run -v`

[runLogs.txt](https://github.com/flutter/flutter/files/7891835/runLogs.txt)

`flutter analyze`

```
Analyzing example...

   info • Use key in widget constructors • lib/main.dart:11:7 • use_key_in_widget_constructors
   info • Avoid `print` calls in production code • lib/main.dart:41:7 • avoid_print
   info • Avoid `print` calls in production code • lib/main.dart:43:9 • avoid_print
   info • Avoid `print` calls in production code • lib/main.dart:72:17 • avoid_print
   info • Avoid `print` calls in production code • lib/main.dart:79:17 • avoid_print

5 issues found. (ran in 1.5s)
```

`flutter doctor -v`

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale en-RO)
    • Flutter version 2.8.1 at /Users/abratule/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (5 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (2 available)
    • X (mobile)   • c79e0258cd07ead15c4c08f8afb6c315246d489b • ios            • iOS 12.5.5 16H62
    • Chrome (web) • chrome                                   • web-javascript • Google Chrome 96.0.4664.55

! Doctor found issues in 2 categories.
```

</details>

One strange thing that I've tried and it kind of worked was to mess around with Podfile.
- removed `pod 'Flutter', :path => 'Flutter'` from `~/flutter/packages/flutter_tools/bin/podhelper.rb`
- in `ios/Podfile` added `pod 'Flutter', '~> 2.5'` (aparently there is a "Flutter" package)
- **Build passed**, the app would get installed but would crash at some point after it opened

**Comment by maheshmnj** [comment]
Hi @andreibratuleanu21, Thanks for filing the issue. Please share your `Podfile` and `pubsec.yaml` file

>  /Users/abratule/workspace/aero_f_ble/example/ios/Runner/Runner-Bridging-Header.h:1:9: note: in file included from /Users/abratule/workspace/aero_f_ble/example/ios/Runner/Runner-Bridging-Header.h:1:
               #import "GeneratedPluginRegistrant.h"
                       ^
               /Users/abratule/workspace/aero_f_ble/example/ios/Runner/GeneratedPluginRegistrant.h:10:9: error: 'Flutter/Flutter.h' file not found
               #import <Flutter/Flutter.h>
                       ^
               1 error generated.
               <unknown>:0: error: failed to emit precompiled header '/Users/abratule/Library/Developer/Xcode/DerivedData/Runner-emmvpmwovcrfsadqbdztkauwfpmu/Build/Intermediates.noindex/PrecompiledHeaders/Runner-Bridging-Header-swift_1IAYIT96INIUM-clang_3MJ7WMOU6EFXF.pch' for bridging header '/Users/abratule/workspace/aero_f_ble/example/ios/Runner/Runner-Bridging-Header.h'
               /Users/abratule/workspace/aero_f_ble/example/ios/Runner/Runner-Bridging-Header.h:1:9: note: in file included from /Users/abratule/workspace/aero_f_ble/example/ios/Runner/Runner-Bridging-Header.h:1:
               #import "GeneratedPluginRegistrant.h"
                       ^
               2 errors generated.

> error: generate-pch command failed with exit code 1 (use -v to see invocation)

**Comment by andreibratuleanu21** [comment]
Sure.

<details>
  <summary>Podfile in `[package]/example/ios`</summary>

```
# Uncomment this line to define a global platform for your project
# platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'development' => :debug,
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!
  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```
</details>


<details>
  <summary>Pubspec.yaml in `[package]`</summary>

```
name: aero_f_ble
description: A new flutter plugin project.
version: 0.0.1
homepage:

environment:
  sdk: ">=2.15.0 <3.0.0"
  flutter: ">=2.5.0"

dependencies:
  flutter:
    sdk: flutter

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^1.0.0

flutter:
  plugin:
    platforms:
      android:
        package: net.bytex.aero_f_ble
        pluginClass: AeroFBlePlugin
      ios:
        pluginClass: AeroFBlePlugin
```
</details>
<details>
  <summary>Pubspec.yaml in `[package]/example`</summary>

```
environment:
  sdk: ">=2.15.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  aero_f_ble:
    path: ../
  cupertino_icons: ^1.0.2
  
dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^1.0.0
flutter:
  uses-material-design: true
```
</details>

**Comment by maheshmnj** [comment]
Thanks for the info @andreibratuleanu21, I suspect this to be related to Xcode, Leaving this issue open for further investigation.  
Duplicate of https://github.com/flutter/flutter/issues/42498 (closed) similar to https://github.com/flutter/flutter/issues/74210 (closed)
cc: @jmagman

**Comment by andreibratuleanu21** [comment]
In this case, if it helps, I have this notification in Xcode (version is: 13.2.1 (13C100))

![Screenshot 2022-01-19 at 16 12 34](https://user-images.githubusercontent.com/43005064/150147981-1fd57379-449c-48b2-9ede-6866890a90af.png)

And clicking update will lead to [this page](https://developer.apple.com/documentation/Xcode/Xcode-Cloud), but no clear instruction of how to get rid of it. Also not sure what does my project has to do with Xcode Cloud.

**Comment by jmagman** [comment]
> - removed `pod 'Flutter', :path => 'Flutter'` from `~/flutter/packages/flutter_tools/bin/podhelper.rb`

PLEASE restore this line, it's what's embedding the Flutter framework into your Flutter app, and your app is now complaining that the framework isn't found.  I can tell from your logs that's what's happening.  Restore the Flutter SDK to the version reflected in your `doctor` or there's no way we can debug this.

> - in `ios/Podfile` added `pod 'Flutter', '~> 2.5'` (aparently there is a "Flutter" package)

Remove this, it isn't correct.

**Comment by andreibratuleanu21** [comment]
Thanks for your input @jmagman, and Sorry, I wasn't clear in my post.

1. At the time I have open this issue that line was already back to it's original form. I was just messing around with some code to help me understand more, but I wouldn't open a issue after having changed things that **should not** be changed.

2. Actually having that line changed made a difference. That's why I mentioned it here. Even if I don't know about most of the things there I can tell that is completely wrong.

**Comment by jmagman** [comment]
1.  I don't see the `Run Script` build phase in your verbose logs, which is what moves the Flutter.framework around.  
<img width="1176" alt="Screen Shot 2022-01-19 at 6 01 34 PM" src="https://user-images.githubusercontent.com/682784/150249948-bab057b1-2396-4d8a-85f2-d5e1fcce4f23.png">

Though it should be there if you recreated your project by removing `example/ios`.  Or did you do that?


2. Can you attach `ios/Runner.xcodeproj/project.pbxproj`?

**Comment by andreibratuleanu21** [comment]
This are my settings in Build Phases:
![Screenshot 2022-01-20 at 12 49 54](https://user-images.githubusercontent.com/43005064/150324814-5556438d-c5cb-40e6-857e-cfe285411bdb.png)
![Screenshot 2022-01-20 at 12 50 11](https://user-images.githubusercontent.com/43005064/150324820-a7cb3ef3-0e04-4c18-bc2d-263ab176545d.png)


And here is my project.pbxproj file:
[project.pbxproj.txt](https://github.com/flutter/flutter/files/7904475/project.pbxproj.txt)

**Comment by andreibratuleanu21** [comment]
The issue is now fixed.

> LATER Edit: Well, I've downloaded Xcode 12.5.1 for nothing because "This version of Xcode [...] is not compatible with macOS Monterey. [...] Download the latest version from the App Store", so I don't think I can downgrade to a major version.

As I've said earlier, I did uninstall Xcode and installed it again. But I don't think that this fixed it.
> 1. I don't see the Run Script build phase in your verbose logs, which is what moves the Flutter.framework around.
> Though it should be there if you recreated your project by removing example/ios. Or did you do that?
> 2. Can you attach example/ios/Runner.xcodeproj/project.pbxproj?

@jmagman I did remove `ios` but I had a backup for my Swift code. I might have accidentally restored `project.pbxproj` file as well. Now, I've deleted ios again an run `flutter create .` and I guess that what fixed the issue for me.

---

TL;DR: if anyone has this issue, make sure to delete `ios` folders, backup only what you need and than `flutter create .` :)

**Comment by maheshmnj** [comment]
>  Now, I've deleted ios again an run flutter create . and I guess that what fixed the issue for me.

Recreating the ios folder would have erased the potential configuration that might have caused this issue, We won't be able to investigate this issue further. In case you face this issue again feel free to file a new issue with a sample project and the requested details in the issue template or comment on this issue and we will reopen it right away.

closing as this issue cannot be diagnosed further.

**Comment by jmagman** [comment]
You had `runOnlyForDeploymentPostprocessing` `Install builds only` on.  Which means that it won't run when you build the app.  This was manually changed, Flutter doesn't touch this build setting.
```
			runOnlyForDeploymentPostprocessing = 1;
			shellPath = /bin/sh;
			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build\n";
```
```
			runOnlyForDeploymentPostprocessing = 1;
			shellPath = /bin/sh;
			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" embed_and_thin\n";
```

> TL;DR: if anyone has this issue, make sure to delete `ios` folders, backup only what you need and than `flutter create .` :)

You should never need to delete the `ios` folder and recreate it unless you've done something like manually edited the Xcode project to not run the scripts that allow Flutter to work.
